lib/charms/mongodb_libs/v0/mongodb_provider.py

```diff
@@ -133,6 +158,144 @@ def oversee_users(self, departed_relation_id: Optional[int], event):
                 logger.info("Drop database: %s", database)
                 mongo.drop_database(database)
 
+    ################################################################################
+    # LEGACY RELATIONS
+    ################################################################################
+    def _on_legacy_relation_created(self, event):
+        """Legacy relations for MongoDB operate without a password and so we update the server accordingly and
+        set a flag.
+        """
+        logger.warning("DEPRECATION WARNING - `mongodb` interface is a legacy interface.")
+
+        # TODO, future PR check if there are any new relations that are related to this charm and will be effected
+        # by loss of password. If so go into blocked state.
+
+        # TODO, future PR check if already running without auth
+        try:
+            self._stop_mongod_service()
+            self._update_mongod_service(auth=False)
+            self._start_mongod_service()
+        except systemd.SystemdError:
+            self.charm.unit.status = BlockedStatus("couldn't restart MongoDB")
+            return
+
+    def _on_legacy_relation_joined(self, event):
+        """
+        NOTE: this is retro-fitted from the legacy mongodb charm: https://launchpad.net/charm-mongodb
+        """
+        logger.warning("DEPRECATION WARNING - `mongodb` interface is a legacy interface.")
+        if not self.charm.unit.is_leader:
+            return
+
+        relation = self.model.get_relation(REL_NAME, event.relation.id)
+
+        # reactive charms set relation data on "the current unit"
+        data = relation.data[self.charm.unit]
+        data["hostname"] = str(self.model.get_binding(PEER).network.bind_address)
+        data["port"] = str(MONGODB_PORT)
+        data["type"] = "database"
+        data["version"] = MONGODB_VERSION
+        data["replset"] = self.charm.app.name
```

Comment:
suggestion to make the above code more readable:

```
updates = {
    "hostname": str(self.model.get_binding(PEER).network.bind_address),
    "port": str(MONGODB_PORT),
     ...
    "replset": self.charm.app.name,
}
relation.data[self.charm.unit].update(updates)
```

Reply:
good idea!